Wikipedia/Code/ArticleViewController+Editing.swift

```diff
@@ -77,23 +103,128 @@ extension ArticleViewController {
     
     private func presentEditor(editorViewController: UIViewController) {
         
-        let navigationController = WMFComponentNavigationController(rootViewController: editorViewController, modalPresentationStyle: .overFullScreen)
-        
-        let needsIntro = !UserDefaults.standard.didShowEditingOnboarding
-        if needsIntro {
-            let editingWelcomeViewController = EditingWelcomeViewController(theme: theme) {
+        let presentEditorAction = {
+            let navigationController = WMFComponentNavigationController(rootViewController: editorViewController, modalPresentationStyle: .overFullScreen)
+            
+            let needsIntro = !UserDefaults.standard.didShowEditingOnboarding
+            if needsIntro {
+                let editingWelcomeViewController = EditingWelcomeViewController(theme: self.theme) {
+                    
+                    self.present(navigationController, animated: true)
+                }
+                editingWelcomeViewController.apply(theme: self.theme)
+                self.present(editingWelcomeViewController, animated: true) {
+                    UserDefaults.standard.didShowEditingOnboarding = true
+                }
+                
+            } else {
                 self.present(navigationController, animated: true)
             }
-            editingWelcomeViewController.apply(theme: theme)
-            present(editingWelcomeViewController, animated: true) {
-                UserDefaults.standard.didShowEditingOnboarding = true
+        }
+        
+        if authManager.permanentUser(siteURL: articleURL) == nil {
+            if authManager.authStateIsTemporary {
+                presentTempEditorSheet()
```

Comment:
So far in these modal cases, you don't have a callback to run the `presentEditorAction` after they have dismissed the modals (Done or Swipe Down). So merging as it is means temporary and anonymous users can never reach editor.

If you want I can merge as-is since it's going in a feature branch and you can fix in a later PR. We have https://phabricator.wikimedia.org/T374015 which could cover fixing this hole, or you can do it as a part of https://phabricator.wikimedia.org/T374012 (I think testing editor for that task will be blocked unless you fix this navigation).